spirv_msl.hpp

```diff
@@ -282,6 +287,14 @@ class CompilerMSL : public CompilerGLSL
 		// Fragment output in MSL must have at least as many components as the render pass.
 		// Add support to explicit pad out components.
 		bool pad_fragment_output_components = false;
+		bool ios_support_base_vertex_instance = false; // Provide the Metal bindings as part of the options structure as that is more convenient.
+		bool ios_use_framebuffer_fetch_subpasses = true; // Use Metal's native frame-buffer fetch API for subpass inputs.
+		bool enforce_storge_buffer_bounds = false; // Storage buffer robustness - clamps access to SSBOs to the size of the buffer
+		
+		bool invariant_float_math = false;
+		
+		bool emulate_cube_array = false; // Emulate texturecube_array with texture2d_array for iOS where this type is not available
+		bool enable_decoration_binding = false; // Allow user to enable decoration binding
```

Comment:
What does this do?

Reply:
`invariant_float_math` enables the use of [`fma`](https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf#page=122) builtin function in Metal.

`emulate_cube_array` adds a function to calculate cube-map-array texture coordinates for a 2D array texture. This is required for iOS versions that do not support cube-map arrays, but 2D array textures.

`enable_decoration_binding` adds the way resource bindings were exposed in a previous revision. Made optional and disabled by default, to not change the behavior.